prusti-common/src/vir/optimizations/folding/expressions.rs

```diff
@@ -349,7 +340,7 @@ impl ast::ExprFolder for ExprOptimizer {
                 new_expr
             }
             _ => {
-                if expr.is_place() {
+                if expr.is_place() { // CMTODO this might be the issue
```

Comment:
Can you elaborate? (:

Reply:
Just a note to myself that I forgot to delete. Some time ago the set of expressions considered as places was very limited (in particular it did not include unfolding).